tesseract/tesseract_urdf/include/tesseract_urdf/collision.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file collision.h
+ * @brief Parse collision from xml string
+ *
+ * @author Levi Armstrong
+ * @date Dec 18, 2017
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2017, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_URDF_COLLISION_H
+#define TESSERACT_URDF_COLLISION_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <tesseract_common/status_code.h>
+#include <Eigen/Geometry>
+#include <tinyxml2.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_scene_graph/utils.h>
+#include <tesseract_urdf/utils.h>
+#include <tesseract_urdf/origin.h>
+#include <tesseract_urdf/geometry.h>
+
+namespace tesseract_urdf
+{
+class CollisionStatusCategory : public tesseract_common::StatusCategory
+{
+public:
+  CollisionStatusCategory() : name_("CollisionStatusCategory") {}
+  const std::string& name() const noexcept override { return name_; }
+  std::string message(int code) const override
+  {
+    switch (code)
+    {
+      case Success:
+        return "Sucessfully parsed 'collision' element";
+      case ErrorParsingOriginElement:
+        return "Error parsing collision 'origin' element!";
+      case ErrorMissingGeometryElement:
+        return "Error missing collision 'geometry' element!";
+      case ErrorParsingGeometryElement:
+        return "Error parsing collision 'geometry' element!";
+      default:
+        return "Invalid error code for " + name_ + "!";
+    }
+  }
+
+  enum
+  {
+    Success = 0,
+    ErrorParsingOriginElement = -1,
+    ErrorMissingGeometryElement = -2,
+    ErrorParsingGeometryElement = -3
+  };
+
+private:
+  std::string name_;
+};
+
+inline tesseract_common::StatusCode::Ptr parse(std::vector<tesseract_scene_graph::Collision::Ptr>& collisions,
+                                               const tinyxml2::XMLElement* xml_element,
+                                               tesseract_scene_graph::ResourceLocatorFn locator)
+{
+  collisions.clear();
+  auto status_cat = std::make_shared<CollisionStatusCategory>();
+
+  // get name
+  std::string collision_name = StringAttribute(xml_element, "name", "");
```

Comment:
Where does this come from? Should the default be empty?

Reply:
I believe so. The name is optional and appears in urdf dom to default to empty.